test/e2e/apps/controller_revision.go

```diff
@@ -153,6 +154,7 @@ var _ = SIGDescribe("ControllerRevision [Serial]", func() {
 		framework.Logf("%s has been patched", patchedControllerRevision.Name)
 
 		ginkgo.By("Create a new ControllerRevision")
+		ds.Spec.Template.Spec.TerminationGracePeriodSeconds = pointer.Int64(1)
```

Comment:
asking without  context, why 1 and why this fix the flake?

Reply:
AIUI the test currently fails because the hash for the daemonset `initialRevision.Data` is the same as the current running daemonset. Changing the spec for `TerminationGracePeriodSeconds` will mean that there is a different hash when the ControllerRevision is created. The number just needs to be different from the default value of 30. Most tests if I'm changing this value I prefer using 1.

Reply:
how was this working earlier ?

Reply:
The current test passes without any issues when using a kind cluster but when It's checked with local-up-cluster it has a hard failure. I don't know the internals enough to know what's the difference between those two cluster setups.

Reply:
the main difference on kind and gce clusters use to be the latency, kind use to be much faster ...

Reply:
Results of the investigation:

The hash is not just computed from the configured values (eg. what you would see in your struct initialization or in the generated YAML), but from all the fields in the DaemonSet struct ([hash.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/util/hash/hash.go)). So if any field changes you would get a different hash even though the final trimmed outputed YAML would stay the same.

This is basically what happened - it was cause by removal  https://github.com/kubernetes/kubernetes/pull/109602 of the `ZZZ_DeprecatedClusterName`  field from object metadata in master (1.25).

So if you connect to a bit older cluster running 1.24 (kind, etc) and you compute a hash from a DaemonSet ([controller_revision.go#L158](https://github.com/ii/kubernetes/blob/6c320b2991c5a9a94b1ac5a618dfec073b584d45/test/e2e/apps/controller_revision.go#L158)) it will differ from the cluster generated hash because the daemon_controller still has has the `ZZZ_DeprecatedClusterName` field, but the client (e2e test) does not have it anymore.

If you connect against the cluster of the same version (eg. `local-up-cluster.sh`) the hashes generated by both sides will be identical and indeed we need this fix that changes the DaemonSet struct a bit to get a different hash for a new revision.